src/services/userActivity.js

```diff
@@ -2,31 +2,228 @@
+  let dailyStats = []
+  const todayKey = today.toISOString().split('T')[0]
+  for (let i = 0; i < 7; i++) {
+    let day = new Date(startOfWeek)
+    day.setDate(startOfWeek.getDate() + i)
+    let dayKey = day.toISOString().split('T')[0]
+
+    let dayActivity = practices[dayKey] ?? null
+    let isActive = dayKey === todayKey
+    let type = (dayActivity !== null ? 'tracked' : (isActive ? 'active' : 'none'))
+    dailyStats.push({ key: i, label: DAYS[i], isActive, inStreak: dayActivity !== null, type, myData: dayKey })
+  }
+
+  return {
+    dailyActiveStats: dailyStats,
+    ... getStreaksAndMessage(practices)
+  }
+}
+
+export async function getUserMonthlyStats(year = new Date().getFullYear(), month = new Date().getMonth(), day = 1) {
+  let data = await userActivityContext.getData()
+  let practices = data?.[DATA_KEY_PRACTICES] ?? {}
+
+  // Get the first day of the specified month and the number of days in that month
+  let firstDayOfMonth = new Date(year, month, 1)
+  let today = new Date()
+
+  let startOfMonth = getMonday(firstDayOfMonth)
+  let endOfMonth = new Date(year, month + 1, 0)
+  let daysInMonth = new Date(year, month + 1, 0).getDate()
+
+  let dailyStats = []
+  for (let i = 0; i < daysInMonth; i++) {
+    let day = new Date(startOfMonth)
+    day.setDate(startOfMonth.getDate() + i)
+    let dayKey = day.toISOString().split('T')[0]
+
+    // Check if the user has activity for the day, default to 0 if undefined
+    let dayActivity = practices[dayKey] ?? null
+    let isActive = dayKey === today.toISOString().split('T')[0]
+    let type = (dayActivity !== null ? 'tracked' : (isActive ? 'active' : 'none'))
+
+    dailyStats.push({
+      key: i,
+      label: dayKey,
+      isActive,
+      inStreak: dayActivity !== null,
+      type,
+    })
+  }
+
+  let filteredPractices = Object.keys(practices)
+    .filter((date) => new Date(date) <= endOfMonth)
+    .reduce((obj, key) => {
+      obj[key] = practices[key]
+      return obj
+    }, {})
+
+  return {
+    dailyActiveStats: dailyStats,
+    ...getStreaksAndMessage(filteredPractices)
+  }
+}
+
+export async function getUserPractices() {
+  let data = await userActivityContext.getData()
+  return data?.[DATA_KEY_PRACTICES] ?? []
+}
+
+export async function recordUserPractice(practiceDetails) {
+  await userActivityContext.update(
+    async function(localContext) {
+      let userData = localContext.data ?? { [DATA_KEY_PRACTICES]: [] }
+      userData[DATA_KEY_PRACTICES].push(practiceDetails)
+      userData[DATA_KEY_LAST_UPDATED_TIME] = Math.round(new Date().getTime() / 1000)
+      localContext.data = userData
+    },
+    async function() {
+      return logUserPractice(practiceDetails)
+    },
+  )
+}
+
+function getStreaksAndMessage(practices)
+{
+  let { currentDailyStreak, currentWeeklyStreak } = calculateStreaks(practices)
+
+  let streakMessage = currentWeeklyStreak > 1
+    ? `That's ${currentWeeklyStreak} weeks in a row! Keep going!`
+    : ''
```

Comment:
I recall that we wanted the daily streak to be shown if exists then the weekly streak? I couldn't find the document on it though. something like this?
```suggestion
  let streakMessage = currentDailyStreak > 0 ? `That's ${currentDailyStreak} Days in a row! Keep going!` :
    currentWeeklyStreak > 1 ? `That's ${currentWeeklyStreak} weeks in a row! Keep going!` : ''
```

Reply:
In my opinion, weekly streaks are more important than daily streaks, but I will put a question. Anyway, I will provide a different response depending on what method(monthly/weekly) is called. For weekly, I need to provide only the streak message and the dailyActive Stats and for monthly, I will provide: dailyActiveStats, weeklyActiveStats, total practice duration, current weekly streak and current daily streak